### ISSUE TITLE
Using libffi to build arm64 library and use it to build node-ffi

### ISSUE THREAD

**Issue by XMegamanX** [node-ffi/node-ffi#393]
Hi all !
I want to use node-ffi on my arm64 device but the libffi embedded in the node-ffi package does not support arm64 whereas the npm libffi module is arm64 compliant. So is there a way to use the cross-compiled libffi to cross-compile node-ffi with arm64  ?

Thanks !

**Comment by k1tzu** [comment]
Yes, the lib included is as old as dinosaurs. You need to get latest libffi from git to deps/libffi 
then cd deps/libffi 
./autogen.sh;./configure --prefix=/usr;make;make install
cp aarch64-unknown-linux-gnu/include/ffi.h /usr/include
cp aarch64-unknown-linux-gnu/include/ffitarget.h /usr/include
cp aarch64-unknown-linux-gnu/fficonfig.h /usr/include
then put https://github.com/node-ffi/node-ffi/blob/master/deps/libffi/libffi.gyp file to deps/libffi with a fix:
line 118
```
      'conditions': [
        ['target_arch=="arm"', {
          'sources': [ 'src/arm/ffi.c' ],
          'conditions': [
            ['OS=="linux"', {
              'sources': [ 'src/arm/sysv.S' ]
            }]
          ]
        }, { # ia32 or x64
```
replace with
```
      'conditions': [
        ['target_arch=="arm64"', {
          'sources': [ 'src/aarch64/ffi.c' ],
          'conditions': [
            ['OS=="linux"', {
              'sources': [ 'src/aarch64/sysv.S' ]
            }]
          ]
        }, { # ia32 or x64
```
go back to module root cd ../../
and node-gyp rebuild

**Comment by jmcomby** [comment]
I fork an arm64 version.
I tried on odroid c2 ubuntu 18.02 aarch64.
Now compiling.
`git clone https://github.com/jmcomby/node-ffi-napi.git`
`cd node-ffi`
`npm install`
`node-gyp rebuild`

**Comment by stiv-yakovenko** [comment]
I am able to use **npm install ffi-napi** on nvidia jetson nano instead of node-ffi.
